carcara/src/drup.rs

```diff
@@ -0,0 +1,331 @@
+use crate::ast::*;
+use indexmap::IndexSet;
+use std::borrow::{Borrow, BorrowMut};
+use std::collections::HashMap;
+use std::hash::{DefaultHasher, Hash, Hasher};
```

Comment:
The CI build is failing because it uses rust version 1.72, which doesn't have `std::hash::DefaultHasher` ([apparently it was added in 1.76](https://blog.rust-lang.org/2024/02/08/Rust-1.76.0.html#stabilized-apis)). Instead, we should use `std::collections::hash_map::DefaultHasher`.

```suggestion
use std::collections::hash_map::DefaultHasher;
use std::collections::HashMap;
use std::hash::{Hash, Hasher};
```